src/lib/features/ai/tools/create-flag.ts

```diff
@@ -0,0 +1,66 @@
+import type { RunnableToolFunctionWithParse } from 'openai/lib/RunnableFunction';
+import type { IAuditUser } from '../../../types';
+import type FeatureToggleService from '../../feature-toggle/feature-toggle-service';
+
+type createFlagOperationalParams = {
+    featureService: FeatureToggleService;
+    auditUser: IAuditUser;
+};
+
+type createFlagParams = {
+    project: string;
+    name: string;
+    description?: string;
+};
+
+type createFlagFunctionParams = createFlagOperationalParams & createFlagParams;
+
+const createFlagFunction = async ({
+    featureService,
+    auditUser,
+    project,
+    name,
+    description,
+}: createFlagFunctionParams) => {
+    try {
+        const response = await featureService.createFeatureToggle(
+            project,
+            { name, description },
+            auditUser,
+        );
+
+        return response;
+    } catch (error) {
```

Comment:
What's the try/catch here for? Debugging?

Reply:
Anything can go wrong when calling this method, so this allows us to forward either the success response or the error message back to the assistant, allowing it to handle both scenarios gracefully.